1.9/networking/service-and-pod-networking.md

```diff
@@ -0,0 +1,422 @@
+
+```json
+    "portDefinitions": [
+        {"port": 2001, "name": "http"}, {"port": 2002, "name": "https"}, {"port": 3000, "name": "mon"}
+    ],
+    "requirePorts" : true
+```
+
+The service and host ports (including the environment variables `$PORT_HTTP`, `$PORT_HTTPS`, and `$PORT_MON`), are both now `2001`, `2002` and `3000`.
+
+Each *port-definition* in a `portDefinitions` array allows you to specify a `protocol`, a `name`, and `labels` for each definition. When starting new tasks, Marathon will pass this metadata to Mesos. Mesos will expose this information in the `discovery` field of the task. Custom network discovery solutions can consume this field.
+
+Example *port-definition* requesting a dynamic `tcp` port named `http` with the label `VIP_0` set to `10.0.0.1:80`:
+
+```json
+    "portDefinitions": [
+        {
+            "port": 0,
+            "protocol": "tcp",
+            "name": "http",
+            "labels": {"VIP_0": "10.0.0.1:80"}
+        }
+    ],
+```
+
+The `port` field is mandatory.
+The `protocol`, `name`, and `labels` fields are optional.
+
+### Referencing Ports
+
+You can reference the *host-port*s in the Dockerfile for our fictitious app as follows:
+
+```bash
+CMD ./my-app --http-port=$PORT_HTTP --https-port=$PORT_HTTPS --monitoring-port=$PORT_MON
+```
+
+Alternatively, if you are not using Docker, or had specified a `cmd` in your service definition, it works in the same way:
+
+```json
+"cmd": "./my-app --http-port=$PORT_HTTP --https-port=$PORT_HTTPS --monitoring-port=$PORT_MON"
+```
+
+## `container` and `container/bridge` Mode
+
+Bridge mode networking allows you to map host ports to ports inside your containers.
+It is particularly useful if you are using a container image with fixed port assignments that you can't modify.
+**Note:** It is not necessary to `EXPOSE` ports in your Dockerfile when using Docker container images.
+
+### Enabling `container/bridge` Mode
+
+Specify `container/bridge` mode through the `networks` property:
+
+```json
+  "networks": [ { "mode": "container/bridge" } ],
+  "container": {
+    "type": "MESOS",
+    "docker": {
+      "image": "my-image:1.0"
+    }
+  },
+```
+
+### Enabling `container` Mode
+
+Specify `container` mode through the `network` property:
+
+```json
+  "networks": [ { "mode": "container", "name": "someUserNetwork" } ],
+  "container": {
+    "type": "MESOS",
+    "docker": {
+      "image": "my-image:1.0"
+    }
+  }
+```
+
+If there is a `--default_network_name` configured for Marathon, then specifying a network name for the `container` network is optional:
+`container` networks with an unspecified (`null`) `name` will inherit the value of the `--default_network_name` flag.
+
```

Comment:
There's a DCOS configuration for controlling the value of this root Marathon command line flag.

It's called `dcos_overlay_network_default_name` as per:
https://github.com/dcos/dcos/blob/ca564b4b90f069b6336cd445e36e9b83ddbabd6c/gen/calc.py#L943

(pretty sure we added this back in 1.9)